src/containerapp/azext_containerapp/containerapp_auth_decorator.py

```diff
@@ -3,11 +3,110 @@
+                self.existing_auth["platform"] = {}
+            self.existing_auth["platform"]["configFilePath"] = self.get_argument_config_file_path()
+
+        if self.get_argument_unauthenticated_client_action() is not None:
+            if "globalValidation" not in self.existing_auth:
+                self.existing_auth["globalValidation"] = {}
+            self.existing_auth["globalValidation"]["unauthenticatedClientAction"] = self.get_argument_unauthenticated_client_action()
+
+        if self.get_argument_redirect_provider() is not None:
+            if "globalValidation" not in self.existing_auth:
+                self.existing_auth["globalValidation"] = {}
+            self.existing_auth["globalValidation"]["redirectToProvider"] = self.get_argument_redirect_provider()
+
+        if self.get_argument_excluded_paths() is not None:
+            if "globalValidation" not in self.existing_auth:
+                self.existing_auth["globalValidation"] = {}
+            self.existing_auth["globalValidation"]["excludedPaths"] = self.get_argument_excluded_paths().split(",")
+
+        self.existing_auth = update_http_settings_in_auth_settings(self.existing_auth, self.get_argument_require_https(),
+                                                                   self.get_argument_proxy_convention(), self.get_argument_proxy_custom_host_header(),
+                                                                   self.get_argument_proxy_custom_proto_header())
+        
     def construct_payload(self):
-        super().construct_payload()
+        self.parent_construct_payload()
+        self.set_up_token_store()
+
+    def set_up_token_store(self):
+        if self.get_argument_token_store() is None:
+            return
+
+        if self.get_argument_token_store() is False:
+            safe_set(self.existing_auth, "login", "tokenStore", "enabled", value=False)
+            return
+
+        safe_set(self.existing_auth, "login", "tokenStore", "enabled", value=True)
+        safe_set(self.existing_auth, "login", "tokenStore", "azureBlobStorage", value={})
+
+        param_provided = 0
+        if self.get_argument_sas_url_secret() is not None:
+            param_provided += 1
+        if self.get_argument_sas_url_secret_name() is not None:
+            param_provided += 1
+        if self.get_argument_blob_container_uri() is not None:
+            param_provided += 1
+
+        if param_provided != 1:
+            raise ArgumentUsageError(
+                'Usage Error: only blob storage token store is supported. --sas-url-secret, --sas-url-secret-name and --blob-container-uri should provide exactly one when token store is enabled')
+        
+        if self.get_argument_blob_container_uri() is not None:
+            safe_set(self.existing_auth, "login", "tokenStore", "azureBlobStorage", "blobContainerUri",
+                    value=self.get_argument_blob_container_uri())
+
+            identity = self.get_argument_blob_container_identity()
```

Comment:
Can customer only update the `--blob-container-identity`?

Reply:
not supported for now